[Docs] End User Doc Page (#18055)
gravitational/teleport: docs/pages/connect-your-client/introduction.mdx

---
title: Introduction to Teleport Clients
description: The basics of connecting to resources with Teleport
---

This guide covers the basics of authenticating to Teleport and
accessing resources. It's written for end-users of resources protected by
Teleport, and includes links to more detailed documentation at the end.

## Client tools

### tsh

> `tsh` lets you authenticate to Teleport, and list and connect to resoruces. After [downloading](/downloads)

Reviewer: ```suggestion
`tsh` lets you authenticate to Teleport and list and connect to resources. After [downloading](/downloads)
```

and installing `tsh`, sign in to your Teleport cluster:

> <ScopedBlock scope={["oss"]}>

Reviewer: Should we add a tab for SSO users to the `oss` `ScopedBlock`? I think OSS Teleport still supports authentication via GitHub.

Author: To address this and the comment below I'm removing the scoped blocks. We'll have just tabbed examples for local and SAML users, and use the Vars component to indicate both user and proxy values.

Reviewer: Sounds good!

<TabItem label="Local user">

```code
$ tsh login --proxy=teleport.example.com --user=alice
Enter password for Teleport user alice:
Tap any security key
> Profile URL:        https://teleport.example.com:443
  Logged in as:       alice
  Cluster:            example.com
  Roles:              access
  Logins:             root, ubuntu, ec2-user
  Kubernetes:         enabled
  Valid until:        2022-11-01 22:37:05 -0500 CDT [valid for 12h0m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
```

</TabItem>
<TabItem label="SAML / OIDC user">

```code
$ tsh login --proxy=teleport.example.com --user=alice@example.com --auth=your-idp-connector
Enter password for Teleport user alice:
Tap any security key
> Profile URL:        https://teleport.example.com:443
  Logged in as:       alice
  Cluster:            example.com
  Roles:              access
  Logins:             root, ubuntu, ec2-user
  Kubernetes:         enabled
  Valid until:        2022-11-01 22:37:05 -0500 CDT [valid for 12h0m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
```

</TabItem>
</Tabs>

</ScopedBlock>

> <ScopedBlock scope={["cloud"]}>

Reviewer: I think you can configure SSO connectors with Teleport Cloud as well.

```code
$ tsh login --proxy=mytenant.teleport.sh --user=alice
Enter password for Teleport user alice:
Tap any security key
> Profile URL:        https://teleport.example.com:443
  Logged in as:       alice
  Cluster:            mytennant.teleport.sh
  Roles:              access
  Logins:             root, ubuntu, ec2-user
  Kubernetes:         enabled
  Valid until:        2022-11-01 22:37:05 -0500 CDT [valid for 12h0m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
```

Depending on how Teleport was configured for your network, you may need
additional flags, such as `--auth=a_saml_connector`. Your administrator should
provide the details required for your particular use case.

</ScopedBlock>

### Teleport Connect

The Teleport Connect app provides all the same access to resources as `tsh` in
a friendly graphic user interface. After [downloading](https://goteleport.com/download/)
and installing Teleport Connect, you can log in and initiate sessions for
Server and Database Access within a single window.

> 1. Click **CONNECT** to connect to the Teleport cluster:

Reviewer: I think we can remove the numbers from this list, since these list items also work as regular paragraphs.

Author: Sure, they work either way. I'm partial to numbered steps since they better indicate order, and IMHO make it easier for a reader switching focus between a doc and the actual product to remember where they were on the doc.

Do you feel strongly about removing them?

Reviewer: For me, it's an aesthetic preference and makes these paragraphs more consistent with those in the rest of the page.  If you have reasons for keeping numbers here, that's okay for me!

   ![A new instance of Teleport Connect](../../img/teleport-connect-init.png)

2. Provide the address of your Teleport Cluster (e.g.
<ScopedBlock scope={["oss", "enterprise"]}>`https://teleport.example.com`</ScopedBlock>
<ScopedBlock scope={["cloud"]}>`https://mytennant.teleport.sh`</ScopedBlock>
) and click **NEXT**.

3. Teleport Connect will ask you for your username, password, and MFA. If
Teleport is integrated with an external Identity Provider (**IdP**), it may
instead prompt you to authenticate in a browser window with that service.

4. Once authenticated, you can browse and connect to all the resources your
user is permitted to access:

   ![An example of Teleport Connect, populated with instances](../../img/use-teleport/connect-ui-overview.png)

### Web UI

> Teleport provides a web interface for users to interact with. This is usually

Reviewer: ```suggestion
Teleport provides a web interface for users to interact with Teleport, e.g., by accessing resources or creating Access Requests. This is usually
```

found at the same URL used to connect to Teleport with (e.g.
<ScopedBlock scope={["oss", "enterprise"]}>`https://teleport.example.com`</ScopedBlock>
<ScopedBlock scope={["cloud"]}>`https://mytenant.teleport.sh`</ScopedBlock>),

> but can be adjusted by Teleport Administrators. The Web UI provides similar

Reviewer: ```suggestion
but you should confirm the Web UI URL with the team that manages your Teleport deployment. The Web UI provides similar
```
since we don't really have a concept of a Teleport Administrator in the docs

access to resources as Telport Connect, and additional access to to Request and
Activity logs for users with the right permissions.

## Protocols
### Server Access (`ssh`)

<Tabs>
<TabItem label="tsh">

`tsh ls` lists the servers you have access to through Server Access (SSH):

```code
$ tsh ls
Node Name           Address         Labels                       
------------------- --------------- ---------------------------- 
server1.example.com 192.0.2.24:3022 access=servers,hostname=server1
server2.example.com 192.0.2.32:3022 access=servers,hostname=server2
```

To connect to a server:

```code
$ tsh ssh ubuntu@server1.example.com
ubuntu@server1:~$
```

</TabItem>
<TabItem label="Teleport Connect">

Under the **Servers** tab, click **CONNECT** next to the server you want to
access. Select or type in a username available to your Teleport user.

In a new tab, Teleport Connect will initiate the connection and provide a
terminal environment.

</TabItem>
<TabItem label="Web UI">

From the <Icon name="server" size="sm" inline/> **Servers** menu, the Teleport
Web UI will list all servers your user has permission to access. The **CONNECT**

> button will open a new tab with a terminal emulator to provide acesss to that server.

Reviewer: ```suggestion
button will open a new tab with a terminal emulator to provide access to that server.
```

Database certificate renewed: valid until 2022-11-02T06:11:50Z [valid for 9h55m0s]
```

</TabItem>
<TabItem label="Teleport Connect">

In the **Databases** tab, select the database you want to access and click
**CONNECT**. Select or type in a username available to your Teleport user.

Teleport Connect will open a tunneled connection to the database. You can copy
the port number and connect with a GUI app at `localhost`, or click **RUN** to
initiate a CLI connection inside Teleport Connect:

![Teleport Connect database connection](../../img/teleport-connect-mysql.png)

</TabItem>
<TabItem label="Web UI">

The Teleport Web UI cannot provide direct connections to databases, but it will
list those that are accessible to your user under <Icon name="database" size="sm" inline/>
**Databases** and provide `tsh` commands to connect from your local terminal environment.

</TabItem>
</Tabs>

### Desktop Access

Desktop Access is available through the Teleport Web UI. In your browser,
navigate to your Teleport cluster (e.g. 
<ScopedBlock scope={["oss", "enterprise"]}>`https://teleport.example.com`</ScopedBlock>
<ScopedBlock scope={["cloud"]}>`https://mytennant.teleport.sh`</ScopedBlock>
).

> 1. From the menu on the right, select <Icon name="desktop" inline size="sm"/> **Desktops**.

Reviewer: Same comment as earlier in the guide re: removing the numbers from the list
